pkg/types/encoder.go

```diff
@@ -23,3 +22,32 @@ func YAMLEncoder(writer io.Writer, v interface{}) error {
 	_, err = writer.Write(buf)
 	return err
 }
+
+func JSONLinesEncoder(writer io.Writer, v interface{}) error {
+	if collection, ok := v.(*GenericCollection); ok {
+		encoder := json.NewEncoder(writer)
+
+		// encode the top level object first
+		err := encoder.Encode(collection.Collection)
+		if err != nil {
+			return err
+		}
+
+		// write collection objects 1 at a time
+		for _, obj := range collection.Data {
+			err = encoder.Encode(obj)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		// if we receive a type that is not a collection fall back to standard json encoding
+		if err := json.NewEncoder(writer).Encode(v); err != nil {
+			return err
+		}
+	}
+
+	// a blank newline at the end indicates the complete response was returned, if this is absent an error occurred in the middle of encoding
+	_, err := writer.Write([]byte("\n"))
```

Comment:
I am uncomfortable with this.

Instead of creating the `encoder` with the `writer` that is passed in, can it be created with a `bytes.Buffer`? That way, the response can be written there and if at any point an error occurs, then the error can returned and a proper status is sent back. If there is no error, then the `bytes.Buffer.Bytes()` is written to the `writer` here.